src/main/java/io/github/pmckeown/dependencytrack/upload/UploadBomMojo.java

```diff
@@ -79,6 +99,40 @@ private String getBomLocation() {
         }
     }
 
+    Optional<ProjectInfo> createProjectInfo() {
```

Comment:
To keep the components targeted and with single responsibility, I'd prefer this was located elsewhere in the code.

Mojo classes are essentially the API interface to invoke functionality from Maven so this feels wrong here.  Also the addition of a lot of new BOM parsing imports are hint too.

This method would ideally be package private to the `project` package as that is the only place that needs the BOM information right now.  All it needs to create the ProjectInfo is the bom location so that could be provided and parsed locally in a helper function or class.